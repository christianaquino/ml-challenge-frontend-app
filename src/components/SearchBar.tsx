import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

interface ISearchBarProps {
  searchText?: string
}

const defaultProps: ISearchBarProps = {
  searchText: '',
};

function SearchBar({ searchText }: ISearchBarProps) {
  const [query, setQuery] = useState(searchText);
  const navigate = useNavigate();

  const handleClick = () => navigate(`/items?search=${query}`);
  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setQuery(target.value);
  };

  return (
    <div>
      <input type="text" onChange={handleChange} />
      <button type="button" onClick={handleClick}>Search</button>
    </div>
  );
}

SearchBar.defaultProps = defaultProps;

export default SearchBar;
