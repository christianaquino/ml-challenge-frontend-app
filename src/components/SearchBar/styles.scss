@use 'sass:map';
@use '../../global.scss' as *;

.searchbar {
  @include grid-12-columns;
  background-color: map.get($colors, 'ml-yellow');
  grid-area: header;
  height: 58px;

  &--container {
    @include grid-12-columns;
    grid-column-start: 2;
    grid-column-end: end;
    align-items: center;
  }

  &--backbutton {
    grid-column: 1;
    text-align: end;
    padding-right: 18px;
  }

  &--logo {
    width: 36px;
  }

  &--textinput {
    font-size: 18px;
    grid-column-start: 2;
    grid-column-end: span 8;
    border-width: 0;
    height: 36px;
    padding-left: 18px;
  }

  &--button {
    padding: 5px;
    grid-column: 10;
    width: 36px;
    height: 38px;
    border-width: 0;
  }
}
