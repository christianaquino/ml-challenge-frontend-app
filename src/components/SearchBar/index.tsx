import React, { useState } from 'react';
import { useNavigate, Link } from 'react-router-dom';

import './styles.scss';

interface ISearchBarProps {
  searchText?: string
}

const defaultProps: ISearchBarProps = {
  searchText: '',
};

function SearchBar({ searchText }: ISearchBarProps) {
  const [query, setQuery] = useState(searchText);
  const navigate = useNavigate();

  const handleClick = () => navigate(`/items?search=${query}`);
  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setQuery(target.value);
  };

  return (
    <header className="searchbar">
      <div className="searchbar--container">
        <Link to="/" aria-label="volver al inicio" className="searchbar--backbutton">
          <img src="/assets/Logo_ML.png" className="searchbar--logo" alt="logo" />
        </Link>
        <input type="text" value={query} onChange={handleChange} placeholder="Nunca dejes de buscar" className="searchbar--textinput" />
        <button type="button" onClick={handleClick} className="searchbar--button" aria-label="buscar">
          <img src="/assets/ic_Search.png" alt="icono buscar" />
        </button>
      </div>
    </header>
  );
}

SearchBar.defaultProps = defaultProps;

export default SearchBar;
