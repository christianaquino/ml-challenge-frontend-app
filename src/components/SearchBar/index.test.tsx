import { render, fireEvent } from '@testing-library/react';
import SearchBar from './index';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
  Link: props => <a { ...props}>{props.children}</a>
}));

describe('SearchBar component', () => {
  test('renders correctly', () => {
    const { container, asFragment } = render(<SearchBar />);

    expect(asFragment()).toMatchSnapshot()
  });

  test('navigates to search results', () => {
    const { container } = render(<SearchBar />);
    const searchInput = container.querySelector('.searchbar--textinput');
    const searchButton = container.querySelector('.searchbar--button');

    fireEvent.change(searchInput, { target: {value: 'iphone'}} )
    fireEvent.click(searchButton)

    expect(mockedUsedNavigate).toHaveBeenCalledWith('/items?search=iphone')
  });
});
