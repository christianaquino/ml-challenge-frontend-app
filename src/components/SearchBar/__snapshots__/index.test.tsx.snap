// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="searchbar"
  >
    <div
      class="searchbar--container"
    >
      <a
        aria-label="volver al inicio"
        class="searchbar--backbutton"
        to="/"
      >
        <img
          alt="logo"
          class="searchbar--logo"
          src="/assets/Logo_ML.png"
        />
      </a>
      <input
        class="searchbar--textinput"
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        aria-label="buscar"
        class="searchbar--button"
        type="button"
      >
        <img
          alt="icono buscar"
          src="/assets/ic_Search.png"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
